TodoList/Tests/TodoList.Services.Tests/Wrappers/DateTimeGeneratorTest.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using TodoList.Contracts.Services;
+using TodoList.Services.Wrappers;
+
+namespace TodoList.Services.Tests.Wrappers
+{
+    [TestFixture]
+    public class DateTimeGeneratorTest
+    {
+        private IDateTimeGenerator _dateTimeGenerator;
+
+        [SetUp]
+        public void Init()
+        {
+            _dateTimeGenerator = new DateTimeGenerator();
+        }
+
+        [Test]
+        public async Task GetCurrentDateTime_ReturnsDelayedDateTimesCorrectly()
+        {
+            var waitingTime = 100;
+            var precisionDeviation = 20;
```

Comment:
Since the `deviation` is this high, it the test takes about 120ms – one of your longest tests. Precision of the Framework is actually about 3-5 ms (cannot find the original article I have this number from, unfortunately), however, `Thread.Sleep` is not meant to be precise (read more about it on [MSDN](https://docs.microsoft.com/en-us/dotnet/api/system.threading.thread.sleep?f1url=https%3A%2F%2Fmsdn.microsoft.com%2Fquery%2Fdev15.query%3FappId%3DDev15IDEF1%26l%3DEN-US%26k%3Dk(System.Threading.Thread.Sleep);k(SolutionItemsProject);k(TargetFrameworkMoniker-.NETFramework,Version%3Dv4.6);k(DevLang-csharp)%26rd%3Dtrue&view=netframework-4.7). If you want a bit more precision, you need to use at least `System.Diagnostics.Stopwatch` – see this [SO](https://stackoverflow.com/a/22223245/1138663) for further guidence (only if you wanted to push the `deviation` to about 3ms, allowing you to lesser the `waitingTime` down to abou 20ms)